TASK: Migrate to PHPStan (adjustments in Flow 9)

mhsdesign opened the pull request:
With https://github.com/neos/flow-development-collection/pull/3218 PHPStan level 1 was added to the whole Flow code base and CI for Flow 8. This upmerged change needs some adjustments to pass the CI in Flow 9

- fix types in code that was introduced with Flow 9
- fix types where neos depends on it (by correcting types and adding `never`)
- adjust unit test as `never` cannot be doubled (eventually this will be fixed via: https://github.com/sebastianbergmann/phpunit/issues/5048)
- fix ci and style as neos 9 followup for https://github.com/neos/flow-development-collection/pull/3218

**Checklist**

- [x] Code follows the PSR-2 coding style
- [ ] Tests have been created, run and adjusted as needed
- [ ] ~The PR is created against the [lowest maintained branch](https://www.neos.io/features/release-roadmap.html)~
- [x] Reviewer - PR Title is brief but complete and starts with `FEATURE|TASK|BUGFIX`
- [x] Reviewer - The first section explains the change briefly for change-logs
- [ ] Reviewer - Breaking Changes are marked with `!!!` and have upgrade-instructions

kitsunet commented:
Mmm, but like in 9.0? I think this will make upmerges harder than they already are with the many chnages in this major version, I would rather do it in 8.0 maybe? In the hopes that we don't upmerge 7.3 all that much anymore...

mhsdesign commented:
Draft as we probably want to target 8.0 https://github.com/neos/flow-development-collection/pull/3218

Things that are not included there:

Adding `never` as return type

kdambekalns commented on Neos.Eel/Classes/ProtectedContext.php:
```diff
@@ -67,31 +67,6 @@ public function getAndWrap($path = null)
-     * @return void
-     * @deprecated Use allow() instead. See https://github.com/neos/flow-development-collection/pull/2024
-     */
-    public function whitelist($pathOrMethods)
```
Sneaky removal… 😎

mhsdesign replied:
yes youre right, separate pr https://github.com/neos/flow-development-collection/pull/3220 :D

kdambekalns commented on Neos.Flow/Classes/Mvc/ActionResponse.php:
```diff
@@ -142,34 +142,6 @@ public function deleteCookie(string $cookieName): void
-     * @return void
-     * @deprecated since Flow 7.0 use setHttpHeader or replaceHttpResponse instead. For now this will still work with $componentClassName of "SetHeaderComponent" or "ReplaceHttpResponseComponent" only.
-     */
-    public function setComponentParameter(string $componentClassName, string $parameterName, $value): void
```
Another sneaky removal…

mhsdesign replied:
yes youre right, separate pr https://github.com/neos/flow-development-collection/pull/3220 :D

kdambekalns commented on .composer.json:
Good point! Should be fixed for Flow 8, too…

mhsdesign replied:
hi sorry i dont understand? seems fine: https://github.com/neos/flow-development-collection/blob/a707af14dfb290423455bd6970d34ee72ddff9c2/.composer.json

kdambekalns reviewed (commented):
Nice, I think. **But** I'd like to see the deprecation removals in their own commit, at least. Otherwise they are camouflaged a bit too much…

mhsdesign committed:
TASK: Migrate to PHPStan (adjustments in Flow 9)

With https://github.com/neos/flow-development-collection/pull/3218 PHPStan level 1 was added to the whole Flow code base and CI for Flow 8. This upmerged change needs some adjustments to pass the CI in Flow 9

- fix types and remove deprecated code
- fix types where neos depends on it (by correcting types and adding `never`)
- adjust unit test as `never` cannot be doubled (eventually this will be fixed via: https://github.com/sebastianbergmann/phpunit/issues/5048)
- fix ci and style as neos 9 followup for https://github.com/neos/flow-development-collection/pull/3218
